examples/http-request-errorbudget.yaml

```diff
@@ -0,0 +1,37 @@
+recordingrule:
+- expr: |
+    sum(label_replace(increase(promhttp_metric_handler_requests_total{namespace="default",job="fooapp"}[30d]), "status_class", "${1}xx", "code", "([0-9])..")) by (status_class)
```

Comment:
Why not do `by (status_class, job, namespace` and then you don't have to add the labels again in the lines below?

Reply:
I think it is just easier to add the labels like that again, coming from a jsonnet perspective. That way I don't need to separate the selectors on some heuristic as much.

Reply:
But aren't you already splitting the selectors into labels (to add them in the `labels:` section)? Adding them there vs. adding the in the `by` clause should be similarly easy.

Another thought: What happens if users provide a regexp selector as `selectors`?